service/history/workflow/update/update.go

```diff
@@ -289,6 +290,13 @@ func (u *Update) OnMessage(
 			return err
 		}
 	}
+
+	// If workflow was completed while processing this WFT, then only Rejection messages can be processed,
+	// because they don't create new events in the history. All other updates must be cancelled.
+	if _, isRejection := msg.(*updatepb.Rejection); !isWorkflowRunning && !isRejection {
```

Comment:
I think it will be easier to understand `!(isWorkflowRunning || isRejection)`, but even better approach is to define explanatory function or variable: `shouldCancel := !(isWorkflowRunning || isRejection)`

Reply:
I tried to leverage go `if` one liner syntax.